### PR TITLE
Add run_name argument to the MLFlowLogger constructor

### LINKED ISSUE #7571: MLFlowLogger can‘t set Run Name
When using MLFlowLogger for lightning model training, we can not set `Run Name` parameters

### PR DESCRIPTION
**Pull request by i-aki-y**
## What does this PR do?

This PR adds a `run_name` argument to the MLFlowLogger constructor.
As is shown in #7571, the `run_name` is used to identify the specific execution (run) in mlflow.
By default, the `run_name` is the same value as the `run_id` that is a UUID.
So, mlflow allows the user to set more informative `run_name` in `mlflow.start_run` method argument^[1].

The current MLFlowLogger does not support setting the `run_name` explicitly.
(As I pointed out in #7571, there is a workaround, but it is hard to notice, I think.)
So, in this PR, the `run_name` argument is added to the MLFlowLogger.

To make this PR, I referred to the current implementation of `mlflow.start_run`^[1]

I would appreciate it if you could review this PR.

[1] https://www.mlflow.org/docs/latest/python_api/mlflow.html#mlflow.start_run

Fixes #7571

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃
👍

### CODE REVIEW

File: pytorch_lightning/loggers/mlflow.py [hunk 1]
```diff
@@ -155,6 +161,14 @@ def experiment(self) -> MlflowClient:
                 )
 
         if self._run_id is None:
+            if self._run_name is not None:
+                self.tags = self.tags or {}
```
Comment (justusschock) [inline pytorch_lightning/loggers/mlflow.py:165]: can we maybe lift this to the init already? So that `self.tags` is always a dict?

**Review by justusschock — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -61,6 +61,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - MLflowLogger now uses the env variable `MLFLOW_TRACKING_URI` as default tracking uri ([#7457](https://github.com/PyTorchLightning/pytorch-lightning/pull/7457))
 
 
+- MLflowLogger now accepts `run_name` as an constructor argument ([#7571](https://github.com/PyTorchLightning/pytorch-lightning/issues/7571))
```
Comment (awaelchli) [inline CHANGELOG.md:64]:
```suggestion
- MLFlowLogger now accepts `run_name` as an constructor argument ([#7622](https://github.com/PyTorchLightning/pytorch-lightning/issues/7622))
```

File: tests/loggers/test_mlflow.py [hunk 3]
```diff
@@ -85,6 +86,33 @@ def test_mlflow_logger_exists(client, mlflow, tmpdir):
...
+    _ = logger.experiment
+    client.return_value.create_run.assert_called_with(experiment_id='exp-id', tags=tags)
+
+    # run_name overrides tags[MLFLOW_RUN_NAME]
+    logger = MLFlowLogger('test', run_name='run-name-1', tags={MLFLOW_RUN_NAME: "run-name-2"}, save_dir=tmpdir)
+    logger = mock_mlflow_run_creation(logger, experiment_id='exp-id')
+    _ = logger.experiment
+    client.return_value.create_run.assert_called_with(experiment_id='exp-id', tags=tags)
+
+    # default run_name (= None) does not append new tag.
```
Comment (awaelchli) [inline tests/loggers/test_mlflow.py:108]:
```suggestion
    # default run_name (= None) does not append new tag
```

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/loggers/mlflow.py (hunk 1 above)
Comment (i-aki-y) [reply pytorch_lightning/loggers/mlflow.py:165]: Thanks. I have a concern about this. The current `self.tags` looks to be a public attribute (it has no prefix '_'), so I worried that the user could set `None` to it after the initialization. 

Note that the document does not say the `tags` are public, so I'm not sure it is intentional: 
https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.loggers.mlflow.html#mlflow-logger

Another minor reason why I put this here is to minimize the side effect. If the `run_name` is not set explicitly, the `self.tags` still be 'None' after this PR. this is the same behavior as the current version.

I could not judge which is better, so any suggestions and comments are welcome!

File: CHANGELOG.md (hunk 2 above)
Comment (i-aki-y) [reply CHANGELOG.md]: Ah, right. Thanks.

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**